src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -1,48 +1,209 @@
+        armMotor2.follow(armMotor1);
+     
+        //armEncoder1.setZeroOffset(offsetRad);
+      
+        //armPID.setTolerance(posToleranceRad, velToleranceRadPSec);
+
+        SmartDashboard.putData("Arm", this);
+
+        //armProfileTimer.start(); <-- don't neeed timer anymore
+
+        setArmTarget(goalState.position);
+
+        // SmartDashboard.putNumber("Arm Max Vel", MAX_FF_VEL );
+        SmartDashboard.putNumber("ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD", ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD);
+        SmartDashboard.putNumber("Arm Tolerance Pos", posToleranceRad);
+        SmartDashboard.putNumber("Arm Tolerance Vel", velToleranceRadPSec);
+    }
+
+    @Override
+    public void periodic() {
+
+        if(DriverStation.isDisabled()) resetGoal();
+
+        //ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD = SmartDashboard.getNumber("ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD", ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD);
+        // armConstraints = new TrapezoidProfile.Constraints(MAX_FF_VEL , MAX_FF_ACCEL );
+        armPID1.setP(kP);
+        armPID1.setI(kI);
+        armPID1.setD(kD);
+        armPID2.setP(kP);
+        armPID2.setI(kI);
+        armPID2.setD(kD);
+
+        //smart dahsboard stuff
+        //SmartDashboard.putBoolean("ArmPIDAtSetpoint", armPID1.atSetpoint());
+        SmartDashboard.putBoolean("ArmProfileFinished", armProfile.isFinished(armProfileTimer.get()));
+        //posToleranceRad = SmartDashboard.getNumber("Arm Tolerance Pos", posToleranceRad);
+        //velToleranceRadPSec= SmartDashboard.getNumber("Arm Tolerance Vel", velToleranceRadPSec);
+
+       // SmartDashboard.putNumber("MaxHoldingTorque", maxHoldingTorqueNM());
+        //SmartDashboard.putNumber("V_PER_NM", getV_PER_NM());
+        SmartDashboard.putNumber("COMDistance", getCoM().getNorm());
+        SmartDashboard.putNumber("InternalArmVelocity", armEncoder.getVelocity());
+        //SmartDashboard.putNumber("Arm Current", armMotor.getOutputCurrent());
+
+       // SmartDashboard.putNumber("ArmPos", getArmPos());
+
+        //driveArm(armProfile.calculate(armProfileTimer.get()));
+
+        autoCancelArmCommand();
+    }
+
+  public void autoCancelArmCommand() {
+        if(!(getDefaultCommand() instanceof ArmTeleop) || DriverStation.isAutonomous()) return;
+
+        double requestedSpeeds = ((ArmTeleop) getDefaultCommand()).getRequestedSpeeds();
+
+        if(requestedSpeeds != 0) {
+            Command currentArmCommand = getCurrentCommand();
+            if(currentArmCommand != getDefaultCommand() && currentArmCommand != null) {
+                currentArmCommand.cancel();
+            }
+        }
+    }
+
+    //#region Drive Methods
+    public void driveArm(double goalAngle){
+      TrapezoidProfile.State goalState = new TrapezoidProfile.State(goalAngle, 0);
+      TrapezoidProfile.State setPoint = armProfile.calculate(kDt, getCurrentArmState(), goalState);
+      double armFeedVolts = armFeed.calculate(goalState.velocity, 0);
```

Comment:
Don't use `getCurrentArmState()` as the initial position. If the PID is slow at moving the arm to the correct setpoint then it will never actually move to the goal. See WPILIB's article on trapezoid profiles.

Reply:
Also wrong feedforward parameters

Reply:
For safety, add code to make it so that if the arm goes beyond a certain threshold to stop moving